Error: Operator called default onErrorDropped when HealthMonitorTask.monitor job can't be started due to termination of a pod

### Expected Behavior

Hi,

We see an error log that **HealthMonitorTask.monitor can't be scheduled**. Based on the exception stack trace and further investigation, it happens rarely and only when k8s pod is terminated.

I would expect that there is no error log as the pod is terminating and it is obvious that scheduler can't be run as threadpools are getting closed.

### Actual Behaviour

The stack trace looks like this

```java
eactor.core.Exceptions$ErrorCallbackNotImplemented: reactor.core.Exceptions$ReactorRejectedExecutionException: Scheduler unavailable
Caused by: reactor.core.Exceptions$ReactorRejectedExecutionException: Scheduler unavailable
....
io.micronaut.management.health.monitor.HealthMonitorTask.monitor(HealthMonitorTask.java:97)
	at io.micronaut.management.health.monitor.$HealthMonitorTask$Definition$Exec.dispatch(Unknown Source)
...
Caused by: java.util.concurrent.RejectedExecutionException: Task java.util.concurrent.FutureTask@47388a61[Not completed, task = io.micrometer.core.instrument.Timer$$Lambda/0x00007f3e3d2afb20@430706d9] rejected from java.util.concurrent.ThreadPoolExecutor@64cc5f34[Terminated, pool size = 0, active threads = 0, queued tasks = 0, completed tasks = 870]
```stack trace

### Steps To Reproduce

_No response_

### Environment Information

java:21

### Example Application

_No response_

### Version

4.3.8

--- a/management/src/main/java/io/micronaut/management/health/monitor/HealthMonitorTask.java
+++ b/management/src/main/java/io/micronaut/management/health/monitor/HealthMonitorTask.java
@@ -88,12 +88,6 @@ public class HealthMonitorTask {
 
         Flux.merge(resultPublishers)
             .collectList()
-            .doOnError(e -> {
-                if (LOG.isErrorEnabled()) {
-                    LOG.error("Health monitor check failed with exception: {}", e.getMessage(), e);
-                }
-                currentHealthStatus.update(HealthStatus.DOWN.describe("Error occurred running health check: " + e.getMessage()));
-            })
             .subscribe(healthResults -> {
                 if (LOG.isTraceEnabled() || LOG.isDebugEnabled()) {
                     healthResults.forEach(healthResult -> {
@@ -115,6 +109,13 @@ public class HealthMonitorTask {
                 } else {
                     currentHealthStatus.update(HealthStatus.UP);
                 }
-            });
+            }, this::onError);
+    }
+
+    private void onError(Throwable e) {
+        if (LOG.isErrorEnabled()) {
+            LOG.error("Health monitor check failed with exception: {}", e.getMessage(), e);
+        }
+        currentHealthStatus.update(HealthStatus.DOWN.describe("Error occurred running health check: " + e.getMessage()));
     }
 }

--- a/management/src/test/groovy/io/micronaut/management/health/monitor/HealthMonitorTaskSpec.groovy
+++ b/management/src/test/groovy/io/micronaut/management/health/monitor/HealthMonitorTaskSpec.groovy
@@ -1,5 +1,8 @@
 package io.micronaut.management.health.monitor
 
+import io.micronaut.health.HealthStatus
+import io.micronaut.management.health.indicator.HealthIndicator
+import io.micronaut.management.health.indicator.HealthResult
 import ch.qos.logback.classic.Level
 import ch.qos.logback.classic.Logger
 import ch.qos.logback.classic.spi.ILoggingEvent
@@ -9,8 +12,16 @@ import io.micronaut.context.annotation.Requires
 import io.micronaut.core.util.StringUtils
 import io.micronaut.runtime.server.EmbeddedServer
 import org.slf4j.LoggerFactory
+import reactor.core.publisher.Hooks
+import reactor.core.publisher.Mono
+import reactor.core.scheduler.Schedulers
 import spock.lang.Specification
 import spock.util.concurrent.PollingConditions
+
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.Executors
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.atomic.AtomicReference
 
 import static java.util.Collections.sort
 
@@ -76,6 +87,40 @@ class HealthMonitorTaskSpec extends Specification {
         logLevel << [Level.INFO, Level.DEBUG, Level.TRACE]
     }
 
+    void "test monitor handles rejected scheduler errors without dropping them"() {
+        given:
+        def currentHealthStatus = Mock(io.micronaut.health.CurrentHealthStatus)
+        AtomicReference<Throwable> droppedError = new AtomicReference<>()
+        CountDownLatch droppedErrorLatch = new CountDownLatch(1)
+        Hooks.onErrorDropped { Throwable throwable ->
+            droppedError.set(throwable)
+            droppedErrorLatch.countDown()
+        }
+
+        def executor = Executors.newSingleThreadExecutor()
+        HealthIndicator indicator = Stub(HealthIndicator) {
+            getResult() >> Mono
+                .fromCallable { HealthResult.builder('schedulerFailure', HealthStatus.UP).build() }
+                .subscribeOn(Schedulers.fromExecutorService(executor))
+        }
+        HealthMonitorTask monitorTask = new HealthMonitorTask(currentHealthStatus, indicator)
+        executor.shutdownNow()
+
+        when:
+        monitorTask.monitor()
+
+        then:
+        1 * currentHealthStatus.update({ HealthStatus status ->
+            status == HealthStatus.DOWN && status.description.orElse('').contains('Scheduler unavailable')
+        })
+        !droppedErrorLatch.await(1, TimeUnit.SECONDS)
+        droppedError.get() == null
+
+        cleanup:
+        Hooks.resetOnErrorDropped()
+        executor.shutdownNow()
+    }
+
     @Requires(property = 'spec.name', value = 'HealthMonitorTask')
     class MemoryAppender extends AppenderBase<ILoggingEvent> {
         List<String> events = []
@@ -86,4 +131,3 @@ class HealthMonitorTaskSpec extends Specification {
         }
     }
 }
-

